app/controllers/assignments_controller.rb

```diff
@@ -577,74 +582,76 @@ def download_config_files
   # Uploads a zip file containing all the files specified in download_config_files
   # and modifies the assignment settings according to those files.
   def upload_config_files
-    error = false
```

Comment:
See below about what I meant about catching errors at the end. This means you don't need to keep track of whether an error occurred or not